source/SkiaSharp.Views/SkiaSharp.Views.WPF/SKGLElement.cs

```diff
@@ -0,0 +1,274 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using OpenTK.Graphics;
+using System.Windows.Media.Media3D;
+using SkiaSharp.Views.Desktop;
+using OpenTK.Wpf;
+using SkiaSharp;
+using OpenTK.Graphics.OpenGL;
+using OpenTK.Platform.Windows;
+using OpenTK;
+using System.Windows.Interop;
+using OpenTK.Platform;
+#if NETCOREAPP || NET
+using OpenTK.Mathematics;
+#endif
+
+namespace SkiaSharp.Views.WPF
+{
+	[DefaultEvent("PaintSurface")]
+	[DefaultProperty("Name")]
+	public class SKGLElement : GLWpfControl, IDisposable
+	{
+		private const SKColorType colorType = SKColorType.Rgba8888;
+		private const GRSurfaceOrigin surfaceOrigin = GRSurfaceOrigin.BottomLeft;
+
+		private bool designMode;
+
+		private GRContext grContext;
+		private GRGlFramebufferInfo glInfo;
+		private GRBackendRenderTarget renderTarget;
+		private SKSurface surface;
+		private SKCanvas canvas;
+
+		private SKSizeI lastSize;
+		private SKGLElementWindowListener listener;
+
+		public SKGLElement()
+			: base()
+		{
+			Initialize();
+		}
+
+		private void Initialize()
+		{
+            designMode = DesignerProperties.GetIsInDesignMode(this);
+            var settings = new GLWpfControlSettings() { MajorVersion = 2, MinorVersion = 1, RenderContinuously = false };
+			
+            this.Render += OnPaint;
+
+            this.Loaded += (s, e) =>
```

Comment:
What happens if we move this control from one control to another. For example you have 2 grids and then you remove from grid A and add to grid B. Will we now have 2 listeners?